framework/src/org/checkerframework/framework/qual/FieldInvariant.java

```diff
@@ -0,0 +1,37 @@
+package org.checkerframework.framework.qual;
+
+import java.lang.annotation.Annotation;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.lang.model.element.TypeElement;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+/**
+ * In the class in which the field invariant holds, then field[i] has annotation[i].
```

Comment:
State the overall purpose.  State that the annotations provided here are strict subtypes of those in the superclass.  Give an example of how to write an annotation.